Mathlib/CategoryTheory/FiberedCategory/BasedCategory.lean

```diff
@@ -0,0 +1,244 @@
+/-
+Copyright (c) 2024 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Lezeau, Calle Sönne
+-/
+
+import Mathlib.CategoryTheory.FiberedCategory.HomLift
+import Mathlib.CategoryTheory.Bicategory.Strict
+import Mathlib.CategoryTheory.Functor.Category
+
+/-!
+# The bicategory of based categories
+
+In this file we define the type `BasedCategory 𝒮`, and give it the structure of a strict
+bicategory. Given a category `𝒮`, we define the type `BasedCategory 𝒮` as the type of categories
+`𝒳` equiped with a functor `𝒳.p : 𝒳 ⥤ 𝒮`.
+
+We also define functors between based categories `𝒳 𝒴 : BasedCategory 𝒮`, via the structure
+`BasedFunctor 𝒳 𝒴`. These are defined as functors between the underlying categories `𝒳.obj` and
+`𝒴.obj` which commute with the projections to `𝒮`.
+
+Natural transformations between based functors `F G : BasedFunctor 𝒳 𝒴` are given by the structure
+`BasedNatTrans F G`. These are defined as natural transformations `α` between the functors
+underlying `F` and `G` such that `α.app a` lifts `𝟙 S` whenever `𝒳.p.obj a = S`.
+-/
+
+universe v₂ u₂ v₁ u₁
+
+open CategoryTheory Functor Category NatTrans IsHomLift
+
+variable {𝒮 : Type u₁} [Category.{v₁} 𝒮]
+
+/-- A based category over `𝒮` is a category `𝒳` together with a functor `p : 𝒳 ⥤ 𝒮`. -/
+@[nolint checkUnivs]
+structure BasedCategory (𝒮 : Type u₁) [Category.{v₁} 𝒮] where
+  /-- The type of objects in a `BasedCategory`-/
+  obj : Type u₂
+  /-- The underlying category of a `BasedCategory`. -/
+  category : Category.{v₂} obj := by infer_instance
+  /-- The functor to the base. -/
+  p : obj ⥤ 𝒮
+
+instance (𝒳 : BasedCategory.{v₂, u₂} 𝒮) : Category 𝒳.obj := 𝒳.category
+
+/-- The based category associated to a functor `p : 𝒳 ⥤ 𝒮`. -/
+def BasedCategory.ofFunctor {𝒳 : Type u₂} [Category.{v₂} 𝒳] (p : 𝒳 ⥤ 𝒮) : BasedCategory 𝒮 where
+  obj := 𝒳
+  p := p
+
+/-- A functor between based categories is a functor between the underlying categories that commutes
+with the projections. -/
+structure BasedFunctor (𝒳 𝒴 : BasedCategory.{v₂, u₂} 𝒮) extends
```

Comment:
I think we should allow `𝒳` and `𝒴` to have different universe parameters. Then, the type of based functors should be equipped with a category structure in this generality. Basically, the data involved in the definitions of the 2-category structure should be more universe polymorphic: the more restricted universe assumption should appear only in the definition of the `Bicategory` instance.

Reply:
Good point!